### ISSUE TITLE
UIAutomator2 selectors in Espresso

### ISSUE THREAD

**Issue by Baterka** [appium/appium#16089]
## Feature request

Add ability to use UIAutomator's UiSelectors and UiScrollable and return Element based on searches with them when using Espresso.

I know that there is limited functionality with mobile commands, but I do not see reason why `findElementByAndroidUIAutomator` cannot be present, when Espresso and UIAutomator2 can be used together.

**Comment by mykola-mokhnach** [comment]
Not really, since these locators return UiObject instances instead of UiObject2. UIA2 driver does several (ugly) hacks to unify these elements for clients. Also, the selectors parser itself is quite a piece of complicated code (basically a small interpreter). But if someone feels lucky to give it a try then feel free to do it.
